charts/ExploreView.tsx

```diff
@@ -88,115 +60,58 @@ export class ExploreView extends React.Component<ExploreProps> {
         const rect = containerNode.getBoundingClientRect()
         const bounds = Bounds.fromRect(rect)
         const store = new RootStore()
-        const model = new ExploreModel()
+        const model = new ExploreModel(store)
+        model.populateFromQueryStr(queryStr)
         return ReactDOM.render(
-            <ExploreView
-                bounds={bounds}
-                model={model}
-                store={store}
-                queryStr={queryStr}
-            />,
+            <ExploreView bounds={bounds} model={model} />,
             containerNode
         )
     }
 
-    model: ExploreModel
-    chart: ChartConfig
-    url: ExploreUrl
-
-    disposers: IReactionDisposer[]
-
-    constructor(props: ExploreProps) {
-        super(props)
-
-        const chartProps = new ChartConfigProps()
-        this.chart = new ChartConfig(chartProps)
-
-        this.model = this.props.model
-
-        this.url = new ExploreUrl(this.model, this.chart.url)
-        this.url.populateFromQueryStr(this.props.queryStr)
-
-        this.disposers = [
-            // We need these updates in an autorun because the chart config objects aren't really meant
-            // to be recreated all the time. They aren't pure value objects and have behaviors on
-            // instantiation that include fetching data over the network. Instead, we rely on their
-            // observable properties, and on this autorun block to connect them to the Explore controls.
-            // -@jasoncrawford 2019-12-04
-            autorun(() => {
-                this.chart.props.type = this.configChartType
-                this.chart.props.hasMapTab = this.isMap
-                this.chart.props.hasChartTab = !this.isMap
-                this.chart.tab = this.isMap ? "map" : "chart"
-            }),
-
-            autorun(() => {
-                if (this.indicatorEntry === null) {
-                    this.chart.update({ dimensions: [] })
-                } else {
-                    const indicator = this.indicatorEntry.entity
-                    if (indicator) {
-                        this.chart.update(chartConfigFromIndicator(indicator))
-                    }
-                }
-            })
-        ]
+    componentWillUnmount() {
+        this.model.dispose()
```

Comment:
This creates a bit of an asymmetry – the model is created outside of the view, but is disposed when the view is removed. It feels like in that case the model should be created by the view itself? That would make the relationship more explicit (one is disposed when the other is disposed).

Reply:
Good point. Creation and disposal are not entirely symmetric… except that creation sets up the autorun. We could put that in a separate “setup” call that the view invokes, for symmetry; or we could do as you suggest and create the model in the view; or we could have the top-level that creates the model also dispose… or we could not bother at all because this is a top-level thing anyway and we're not creating and destroying them all the time.

Reply:
After having played with this a bit and tried it different ways, I'm inclined to either just leave it as-is despite the asymmetry, or to remove the componentWillUnmount/dispose call completely and maybe make a note in the bootstrap call that there's no need to dispose of the model because it lasts the lifetime of the window.

I'm leaving it as is for now but let me know if you have more input or strong preferences here

Reply:
:+1: Yeah I agree it doesn't matter for now, can stay as it is.